website/docs/feature-flag-tutorials/dotnet/dotnet-examples.md

```diff
@@ -0,0 +1,397 @@
+
+### How to do canary deployments with a feature flag in .NET?
+
+Canary deployments help find abnormalities, and align with the agile process for faster releases. And quick reversions, if necessary.
+
+Unleash has a few ways to help manage canary deployments for .NET apps at scale:
+
+-   Using a [gradual rollout](/reference/activation-strategies#gradual-rollout) (which we implemented in the [previous section](#gradual-rollouts-for-.net-apps)) would be a simple use case but would reduce the amount of control you have over who gets the new feature.
+
+-   [Strategy constraints](/reference/strategy-constraints) or [segments](/reference/segments) (which are a collection of constraints) to determine which user receives which version for more control than a gradual rollout.
+
+-   [Strategy variants](/reference/strategy-variants) for more advanced use cases. For example, if you want to test 2 different versions of a feature, you can use a strategy variant to split your population of users and conduct an A/B test with them.
+
+Let’s walk through how to utilize **strategy constraints** in our .NET app.
+
+### Configure strategy constraints for canary deployments
+
+We will build a strategy constraint on our existing gradual rollout strategy. This will allow us to target a subset of users for the rollout.
+
+In Unleash, start from the feature flag view and edit your Gradual Rollout strategy from your development environment.
+
+![Edit your gradual rollout strategy by selecting the 'edit strategy' button.](/img/ex-edit-strategy.png)
+
+This will take you to the gradual rollout form. Next, add a new constraint.
+
+![Add a constraint to your strategy.](/img/ex-add-constraint.png)
+
+Let’s say we are experimenting with releasing the “delete” feature to all environments except production to test it with internal users before releasing it to all users.
+
+![The new constraint form includes a context field, operator, and values field to customize the conditions under which a user will be exposed to the flag](/img/ex-constraint-page.png)
+
+We can configure the constraint in the form to match these requirements:
+
+-   The context field is set to **environment**
+-   The operator is set to **NOT_IN**
+-   The value added is **production**
+
+Once you’ve filled out the proper constraint fields, select ‘Done’ and save the strategy.
+
+![Once you add a constraint, you can see it listed underneath your strategy in Unleash.](/img/python-ex-constraint-added.png)
+
+Your release process is now configured with an environment-dependent strategy constraint. Since we've set the rollout to 100%, the feature will be released to all users that are not in the `production` environment.
+
+Alternatively, you can send an API command to apply the same requirements:
+
+```csharp
+HttpClient client = new HttpClient();
+
+string url = "<your-unleash-url>/api/admin/projects/:projectId/features/:featureName/environments/:environment/strategies/:strategyId";
```

Comment:
```suggestion
string url = $"{unleashUrl}/api/admin/projects/{projectId}/features/{featureName}/environments/{environment}/strategies/{strategyId}";
```
Perhaps do something like this for these urls so if people copy+paste then they get build errors for missing parameters instead of weird responses from Unleash.